extensions/amp-social-share/0.1/validator-amp-social-share.protoascii

```diff
@@ -72,11 +54,31 @@ tags: {  # <amp-social-share>
   attrs: {
     name: "type"
     mandatory: true
-    value_regex: "(twitter|facebook|pinterest|linkedin|gplus|email)"
+  }
+  attrs: {
+    name: "data-share-endpoint"
+    value_url: {
+      allowed_protocol: "ftp"
+      allowed_protocol: "http"
+      allowed_protocol: "https"
+      allowed_protocol: "mailto"
+      # Whitelisting additional commonly observed third party
+      # protocols which should be safe
+      allowed_protocol: "sms"
+      allowed_protocol: "viber"
+      allowed_protocol: "whatsapp"
```

Comment:
@dvoytenko I want to get your opinion on this as well. @powdercloud and I chatted about this, this is mainly to blacklist bad protocols (I know `javascript:`, not sure what else). But blacklisting is not possible currently in the validator. 

The problem with this is that it really limits what users can use until they add them to the validator, which I am not really a big fan of. Should we just drop the whitelisting in the validator and enforce it on runtime?